presto-cassandra/src/main/java/com/facebook/presto/cassandra/NativeCassandraSession.java

```diff
@@ -253,6 +268,33 @@ public CassandraTable getTable(SchemaTableName schemaTableName)
 
     private KeyspaceMetadata getKeyspaceByCaseInsensitiveName(String caseInsensitiveSchemaName)
             throws SchemaNotFoundException
+    {
+        try {
+            return keyspaceCache.get(caseInsensitiveSchemaName);
+        }
+        catch (Exception e) {
+            if (e.getCause() != null) {
+                Throwable cause = e.getCause();
+                if (cause instanceof SchemaNotFoundException) {
+                    throw (SchemaNotFoundException) cause;
+                }
+
+                if (cause instanceof PrestoException) {
+                    throw (PrestoException) cause;
+                }
+            }
+
+            if (RuntimeException.class.isAssignableFrom(e.getClass())) {
+                throw (RuntimeException) e;
+            }
+            else {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private KeyspaceMetadata getKeyspaceByCaseInsensitiveName0(String caseInsensitiveSchemaName)
+            throws SchemaNotFoundException
     {
         List<KeyspaceMetadata> keyspaces = executeWithSession(session -> session.getCluster().getMetadata().getKeyspaces());
```

Comment:
This line performs a remote call to Cassandra server which is guaranteed to get the latest data back. If KeyspaceMetadata is cached for one minute, how do you make sure the data is still valid?

Reply:
There is risk that we might use staled data, I checked the usage of returned KeyspaceMetadata, which is used to 

* get the keyspace name
* list the tables in the keyspace
* to check whether a specified table is Materialized View

So from my point of view the side effect of caching the keyspace is not so severe, since the keyspace name/the table list does not change so often.

The reason I want to cache it is described in this issue: https://github.com/prestodb/presto/issues/16103, current code will do a remote call for column, which I don't think is necessary.

Reply:
LGTM. need someone one from FB to take a look.